model/models.py

```diff
@@ -152,6 +154,91 @@ def forward(self, sub, rel):
 		score = torch.sigmoid(x)
 		return score
 
+
+	# Documentation
+
+
+class CompGCN_CTKGC(CompGCNBase):
+		"""
+		Implements the CTKGC Scoring Function as in the paper https://link.springer.com/article/10.1007/s10489-021-02438-8 and writes to the console the total parameter used in the network
+
+		The scoring Function works by
+			1. Build the Entity-relation matrix by multiplying sub_emb.transpose(1.0) and rel_emb
+			2. Then we perform a Convolution on Entity-relation matrix
+			3. We Aggregate all the obtained feature,
+			4. Convert the above feature maps to a 1D Vector
+			5. We project our 1D vector onto a candidate objects to obtain the predicted object embedding
+				1. We do 5. for all our entities (candidate objects)
+			6. Use the sigmoid function to calculate the score of the object embedding
+		"""
+		def __init__(self, edge_index, edge_type, params=None):
+			"""Init ComGCN Base and all the layers used in the Scoring function CTKGC"""
+			super(self.__class__, self).__init__(edge_index, edge_type, params.num_rel, params)
+
+			self.drop = torch.nn.Dropout(self.p.hid_drop)
+			self.hid_drop0 = torch.nn.Dropout(self.p.hid_drop)
```

Comment:
Why is there a zero in the parameter name? I think it would be better readable, if the zero is deleted, where possible.

Reply:
removed zeros